chapter/background.md

```diff
@@ -0,0 +1,607 @@
+# Background
+
+This thesis illustrates an approach of implementing a language server for the Nickel language which communicates with its clients, i.e. editors, over the open Language Server Protocol (in the following abbreviated as *LSP*).
+The current chapter provides the background on the technological details of the project.
+As the work presented aims to be transferable to other languages using the same methods, this chapter will provide the means to distinguish the nickel specific implementation details.
+
+The primary technology built upon in this thesis is the language server protocol.
+The first part of this chapter introduces the LSP, its rationale and improvements over classical approaches, technical capabilities and protocol details. 
+The second part is dedicated to Nickel, elaborating on the context and use-cases of the language followed by an inspection of the technical features of Nickel.
+
+## Language Server Protocol
+
+Language servers are today's standard of integrating support for programming languages into code editors.
```

Comment:
First define, before telling that they are standard or done by X. 

Language Servers are XXXXXX

Do you need to define / make a difference between "language server" and "language server protocol"? clarify.

Reply:
https://github.com/ysndr/MSc-Thesis/pull/27/files#r855145634